staging/src/k8s.io/apiserver/pkg/authorization/authorizer/interfaces.go

```diff
@@ -76,6 +76,12 @@ func (f AuthorizerFunc) Authorize(a Attributes) (bool, string, error) {
 	return f(a)
 }
 
+// RuleResolver provides a mechanism for resolving the list of rules that apply to a given user within a namespace.
+type RuleResolver interface {
+	// RulesFor get the list of cluster wide rules, the list of rules in the specific namespace, incomplete status and errors.
+	RulesFor(user user.Info, namespace string) ([]ResourceRuleInfo, []NonResourceRuleInfo, bool, error)
```

Comment:
It seems weird to return `false` when everything's okay. I don't know how much we want to worry about this for an internal interface, particularly if it matches up to the external API, but Go methods usually return a boolean that's `true` to indicate success.

Reply:
@ericchiang we return complete status instead incomplete here?

Reply:
Or just infer "incomplete" from the error and not return a boolean.

I don't think it matters a ton, but the current bool is kinda weird.